frontend/apps/reader/modules/readeruserhyph.lua

```diff
@@ -0,0 +1,248 @@
+local DataStorage = require("datastorage")
+local Event = require("ui/event")
+local FFIUtil = require("ffi/util")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local lfs = require("libs/libkoreader-lfs")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local ReaderUserHyph = WidgetContainer:new{
+    -- return values from setUserHyphenationDict (crengine's UserHyphDict::init())
+    USER_DICT_RELOAD = 0,
+    USER_DICT_NOCHANGE = 1,
+    USER_DICT_MALFORMED = 2,
+    USER_DICT_ERROR_NOT_SORTED = 3,
+}
+
+-- returns path to the user dictionary
+function ReaderUserHyph:getDictionaryPath()
+    return FFIUtil.joinPath(DataStorage:getSettingsDir(),
+        "user-" .. tostring(self.ui.document:getTextMainLangDefaultHyphDictionary():gsub(".pattern$", "")) .. ".hyph")
+end
+
+-- Load the user dictionary suitable for the actual language
+-- if reload==true, force a reload
+-- Unload is done automatically when a new dictionary is loaded.
+function ReaderUserHyph:loadDictionary(name, reload)
+    if G_reader_settings:isTrue("hyph_user_dict") and lfs.attributes(name, "mode") == "file" then
+        local ret = self.ui.document:setUserHyphenationDict(name, reload)
+        -- this should only happen, if a user edits a dictionary by hand or the user messed
+        -- with the dictionary file by hand. -> Warning and disable.
+        if ret == self.USER_DICT_ERROR_NOT_SORTED then
+            UIManager:show(InfoMessage:new{
+                text = T(_("The user dictionary\n%1\nis not alphabetically sorted.\n\nIt will not be used."), name),
+            })
+            logger.warn("UserHyph: Dictionary " .. name .. " is not sorted alphabetically.")
+            G_reader_settings:makeFalse("hyph_user_dict")
```

Comment:
`It will not be used.` => `It has been disabled.` ? as it's what you actually do with this makeFalse.